### PR TITLE
Fixed issue #74

### LINKED ISSUE #74: [Cyril] [Functionality] Minigames are no longer accessible after returning to the main menu
**Description**
When the user completes a minigame and returns to the main menu, they will notice the minigames are no longer accessible.

**Steps to reproduce**
1. Boot up Dev.0.0.3.0. Pre-Release
2. Tap on Dog
3. Tap on minigames Feeding/Agility
4. Complete Minigame in anyway (win/lose)
5. Tap return to Main Menu button
6. Tap on Dog

Expected behavior:
User is still able to access minigames.

Actual behavior:
Unable to access minigames 

**Reproduced on:**
Device: Samsung Galaxy A54
OS: Android 13
Browser N/A
Version Dev 0.0.3.0 Pre-Release

**Please see Attachment below**

https://github.com/abdrdg/Foundry/assets/134995217/20a37b96-c088-48e8-bf7d-27ea16484fa7

### PR DESCRIPTION
**Pull request by johnzillakaiju**
Added a DestroyThySelf function in the singleton instance. pls call this function after loading the dogso in the mini game so that going back to the main menu it will have a fresh instance and won't cause any weird behavior. also added a savedogdata at the start function of getdogso so that console errors regarding saved data missing won't occur

### CODE REVIEW

**Review by Syoraka — APPROVED** [review]
Approved. The game no longer enters a soft lock state after going back to Main Menu after a minigame.

**Review by abdrdg — APPROVED** [review]
Good description! Fix verified. However, found an issue within the grooming scene where the user gets stuck on the second playthrough of the grooming mini-game.

File: Assets/Assets/Scripts/DataManager/ScriptableObjects/GetDogSO.cs
Comment (abdrdg) [inline Assets/Assets/Scripts/DataManager/ScriptableObjects/GetDogSO.cs:1]: Good job with the comments @johnzillakaiju

Comment (johnzillakaiju) [reply Assets/Assets/Scripts/DataManager/ScriptableObjects/GetDogSO.cs:1]: oh never touched the grooming scene earlier ): might be a different issue

**Pull request merged**